File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s612878490_commented.go
import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	firstLine := scanner.Text()
	parts := strings.Split(firstLine, " ")
	N, _ := strconv.Atoi(parts[0])
	M, _ := strconv.Atoi(parts[1])

	shortage := make([]int, M)
	for i := 0; i < M; i++ {
		scanner.Scan()
		line := scanner.Text()
		A, _ := strconv.Atoi(line)
		if N > A {
			shortage[i] = N - A
		} else {
			shortage[i] = 0
		}
	}

	sort.Ints(shortage)

	ans := 0
	for i := 0; i < len(shortage)-1; i++ {
		ans += shortage[i]
	}

	fmt.Println(ans)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s612878490_commented.py
import sys

line = sys.stdin.readline().split()
N = int(line[0])
M = int(line[1])

shortage = []
for _ in range(M):
    A = int(sys.stdin.readline())
    shortage.append(max(0, N - A))

shortage.sort()

ans = sum(shortage[:-1])

print(ans)
# End of Code
